Gems/GradientSignal/Code/Source/Editor/EditorImageGradientComponent.cpp

```diff
@@ -57,14 +73,144 @@ namespace GradientSignal
         BaseClassType::Activate();
         m_paintBrush.Activate(AZ::EntityComponentIdPair(GetEntityId(), GetId()));
 
+        AZ::Data::AssetBus::Handler::BusConnect(m_configuration.m_imageAsset.GetId());
+        AzToolsFramework::PaintBrushComponentRequestBus::Handler::BusConnect(GetEntityId());
+
         m_componentModeDelegate.ConnectWithSingleComponentMode<EditorImageGradientComponent, EditorImageGradientComponentMode>(
             AZ::EntityComponentIdPair(GetEntityId(), GetId()), nullptr);
     }
 
+    void EditorImageGradientComponent::OnAssetReady(AZ::Data::Asset<AZ::Data::AssetData> asset)
+    {
+        if (!m_configuration.m_overrideAsset.Get())
+        {
+            AZ::Data::AssetId newAssetId = AZ::Data::AssetId(AZ::Uuid::CreateRandom());
+            AZ::Data::Asset<ImageAsset> imageAsset = asset;
+            m_configuration.m_overrideAsset = AZ::Data::AssetManager::Instance().CreateAsset(
+                newAssetId, asset.GetType(), m_configuration.m_overrideAsset.GetAutoLoadBehavior());
+
+            m_component.CopyImageAsset(imageAsset, m_configuration.m_overrideAsset);
+        }
+    }
+
     void EditorImageGradientComponent::Deactivate()
     {
         m_componentModeDelegate.Disconnect();
+
+        AZ::Data::AssetBus::Handler::BusDisconnect();
+        AzToolsFramework::PaintBrushComponentRequestBus::Handler::BusDisconnect();
+
         m_paintBrush.Deactivate();
         BaseClassType::Deactivate();
     }
+
+    void EditorImageGradientComponent::OnCompositionChanged()
+    {
+        BaseClassType::OnCompositionChanged();
+
+        if (!AZ::Data::AssetBus::Handler::BusIsConnectedId(m_configuration.m_imageAsset.GetId()))
+        {
+            m_configuration.m_useOverride = false;
+            m_configuration.m_overrideAsset = { AZ::Data::AssetLoadBehavior::QueueLoad };
+            m_path = "";
+
+            m_component.ClearOverrideConfiguration();
+
+            AZ::Data::AssetBus::Handler::BusDisconnect();
+            AZ::Data::AssetBus::Handler::BusConnect(m_configuration.m_imageAsset.GetId());
+        }
+    }
+
+    void EditorImageGradientComponent::SavePaintLayer()
+    {
+        char projectPath[AZ_MAX_PATH_LEN];
+        AZ::IO::FileIOBase::GetInstance()->ResolvePath("@devassets@", projectPath, AZ_MAX_PATH_LEN);
```

Comment:
There are `ResolvePath` function overloads that can return or accept an AZ::IO::FixedMaxPath
```
virtual bool ResolvePath(AZ::IO::FixedMaxPath& resolvedPath, const AZ::IO::PathView& path) const = 0;
AZStd::optional<AZ::IO::FixedMaxPath> ResolvePath(const AZ::IO::PathView& path) const;
```

Furthermore this isn't the proper way to query the projectPath.

The proper way is to invoke `AZ::Utils::GetProjectPath()`
```suggestion
        AZ::IO::FixedMaxPath = AZ::Utils::GetProjectPath()
```